### PR TITLE
changed assertion styling for test_Restriction.py

### PR DESCRIPTION
**Pull request by lacrosse91**
- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file, have run ``pre-commit`` locally,
and understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

This pull request addresses issue #2835

### CODE REVIEW

File: Tests/test_Restriction.py
Comment (peterjc) [inline Tests/test_Restriction.py:234]: This and the line above look like the problem cases, and I think that neither ``self.assertEqual`` nor ``self.assertNotEqual`` will work here? I've not tried that, but I did look at the source code:

https://github.com/python/cpython/blob/master/Lib/unittest/case.py

However, I think the originals using ``self.assertFalse`` are actually clearer than ``self.assertEqual(..., False)``.

I'm going to ask for a second opinion, but I would be inclined to leave these two lines as they are. We could add the special comment ``# noqa: A500`` to tell flake8 to ignore the warning from flake8-assertive (if installed).

@MarkusPiotrowski could you comment?

Comment (lacrosse91) [reply Tests/test_Restriction.py:234]: Thanks for review.
Should I wait for his response?

Comment (peterjc) [reply Tests/test_Restriction.py:234]: Wait for now - the second opinion might be different.

Comment (MarkusPiotrowski) [reply Tests/test_Restriction.py:234]: `Bio.Restriction` overwrites the `==` and `!=` comparisons for some special meaning regarding neo- and isoschizomers. So what is actually tested here are the comparison operators and thus I think that the original tests are indeed clearer.

Comment (peterjc) [reply Tests/test_Restriction.py:234]: Thanks Markus.

@lacrosse91 could you revert those two lines (and optionally add the ``# noqa: A500`` comment - which should make flake8-assertive happy and document this use of ``assertTrue`` is deliberate).

**Review by peterjc — CHANGES_REQUESTED** [review]
See comments about the ``Acc65I == Asp718I`` and ``Acc65I != Asp718I`` tests.

**Comment by lacrosse91** [comment]
@peterjc and @MarkusPiotrowski 
Thank you for advice.
please review the changes.

**Review by peterjc — APPROVED** [review]

**Pull request merged**
